edxval/api.py

```diff
@@ -428,21 +428,30 @@ def import_from_xml(xml, edx_video_id, course_id=None):
     if xml.tag != 'video_asset':
         raise ValCannotCreateError('Invalid XML')
 
-    if Video.objects.filter(edx_video_id=edx_video_id).exists():
+    try:
+        video = _get_video(edx_video_id)
```

Comment:
_get_video does unnecessary prefetching. It might be best to add a prefetch argument to _get_video.